include/nuttx/fs/fs.h

```diff
@@ -209,13 +211,19 @@ struct file_operations
   int     (*close)(FAR struct file *filep);
   ssize_t (*read)(FAR struct file *filep, FAR char *buffer, size_t buflen);
   ssize_t (*write)(FAR struct file *filep, FAR const char *buffer,
-                   size_t buflen);
+                    size_t buflen);
   off_t   (*seek)(FAR struct file *filep, off_t offset, int whence);
   int     (*ioctl)(FAR struct file *filep, int cmd, unsigned long arg);
+  int     (*truncate)(FAR struct file *filep, off_t length);
```

Comment:
why char driver need truncate callback? should we remove FIOC_TRUNCATE directly?

Reply:
I am not sure what you mean; there are many cases where you can utilize truncate. In our case we use it to implement posix shm, where you set the size of the shared memory area with truncate. In general, it is used to set the size of a file, memory buffer or whatever the device node represents.

Reply:
Ok, I understand that you want to implement ftruncate for shm:
https://man7.org/linux/man-pages/man7/shm_overview.7.html
I would suggest you move truncate related change to a new PR, so we can merge it quickly. BTW, it's better that the new PR contain two patch:
1.git revert dbc163f1b0b78543bb93e6e35142ec2061db85d6
2.Add  truncate callback.

Reply:
I'd prefer doing it in this same. There are many file_operations still which need to be modified, I don't want to spend time doing those many times.

Can you please be more specific why the truncate should be done separately? I am in no hurry with truncate, I won't need it anyhow without mmap and munmap.

Reply:
> I'd prefer doing it in this same. There are many file_operations still which need to be modified, I don't want to spend time doing those many times.
> 

Ok, you can have all patches in one PR like this first:

1. git revert dbc163f
2. Add truncate callback
3. Add mmap/munmap callback

After this, you can create a new PR just include the first two patch. Once the new PR get merged, you can simply rebase this first PR.

> Can you please be more specific why the truncate should be done separately?

Since

1. It isn't related to mmap/munmap directly
2. The change is very simple and can merge quickly

> I am in no hurry with truncate, I won't need it anyhow without mmap and munmap.

If so, it's fine to keep all in one PR, but it's still good if you can split truncate change from mmap since it's always good that one patch do one thing.

Reply:
fine...

Reply:
done